# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/test/TestCookie.cpp
# Commit: 532731e94bb2
# Full Hash: 532731e94bb23b01b5e7136bee829844a0f08bc2
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-06-01 21:42:28
# Regressor Bug: 1638358
# File Overlap Count: 1
# Description:
#   Bug 1638358 - Cookie Schemeful Same-Site - part 5 - schemeful comparison, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75629
# ==============================================================================

diff -r fad2ba760157 -r 532731e94bb2 netwerk/test/TestCookie.cpp
--- a/netwerk/test/TestCookie.cpp	Mon Jun 01 11:43:30 2020 +0000
+++ b/netwerk/test/TestCookie.cpp	Mon Jun 01 11:43:28 2020 +0000
@@ -199,6 +199,7 @@
   Preferences::SetBool("network.cookie.sameSite.laxByDefault", false);
   Preferences::SetBool("network.cookieJarSettings.unblocked_for_testing", true);
   Preferences::SetBool("dom.securecontext.whitelist_onions", false);
+  Preferences::SetBool("network.cookie.sameSite.schemeful", false);
 }
 
 TEST(TestCookie, TestCookieMain)
@@ -725,6 +726,7 @@
   GetACookieNoHttp(cookieService, "https://www.security.test/", cookie);
   EXPECT_TRUE(CheckResult(cookie.get(), MUST_EQUAL,
                           "test_modify_cookie=non-security-cookie"));
+
   // Test the non-security cookie can set when domain or path not same to secure
   // cookie of same name.
   SetACookie(cookieService, "https://www.security.test/", "test=security3");
@@ -735,6 +737,11 @@
   GetACookieNoHttp(cookieService, "http://www.security.test/", cookie);
   EXPECT_TRUE(CheckResult(cookie.get(), MUST_CONTAIN, "test=non-security2"));
 
+  Preferences::SetBool("network.cookie.sameSite.schemeful", true);
+  GetACookieNoHttp(cookieService, "http://www.security.test/", cookie);
+  EXPECT_FALSE(CheckResult(cookie.get(), MUST_CONTAIN, "test=security3"));
+  Preferences::SetBool("network.cookie.sameSite.schemeful", false);
+
   // *** nsICookieManager interface tests
   nsCOMPtr<nsICookieManager> cookieMgr =
       do_GetService(NS_COOKIEMANAGER_CONTRACTID, &rv0);